### ISSUE TITLE
Failing to authenticate with Azure Artifacts on premise

### ISSUE THREAD

**Issue by Skoucail** [PowerShell/PSResourceGet#926]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

I'm trying to Install-PSResource a module provided by a private nuget repository hosted on a Azure Devops Server 2020.

I installed the Microsoft.CredentialProvider. I checked if it works with nuget.exe and a network monitoring tool (fiddler).
With nuget exe i can access the repo without issues. And i see in the HTTP calls nuget.exe correctly added an authentication header "Authentication BASIC -TOKEN-".

But from powershell i keep getting "Error retrieving resource from repository: Unable to load the service index for source https://-private-server-url-/-collectionname-/-project id-/_packaging/PowerShellGallery/nuget/v3/index.json

The steps i'm trying are pretty basic.
First did register the repo without issues with Register-PSResourceRepository. 
Then i try to install the module with Install-PSResource and i receive the error.

In the network monitoring tool i only see calls without authentication headers or calls with NTLMSSP authentication. Never a call with the BASIC authentication. The client used for the NuGet calls is: NuGet Client V3/6.2.2

Can someone tell my why it looks like the Microsoft.CredentialProvider seems to be ignored?
Or how i can enable verbose logging for the NuGet calls?

### Expected behavior

```console
Install-PSResource can install modules from a private Azure Devops Artifact Repository
```

### Actual behavior

```console
Install-PSResource fails with error "Error retrieving resource from repository: Unable to load the service index for source https://<private-server-url>/<collectionname>/<project id>/_packaging/PowerShellGallery/nuget/v3/index.json"
```

### Error details

```console
Error retrieving resource from repository: Unable to load the service index for source https://<private-server-url>/<collectionname>/<project id>/_packaging/PowerShellGallery/nuget/v3/index.json
```

### Environment data

```powershell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     3.0.19     beta19     PowerShellGet                       {Find-PSResource, Get-PSResource, Get-PSResourceR…
```

**Comment by Skoucail** [comment]
To be complete, before trying all that, i tried with the -Credential option.
Same thing, no calls were made with a Authentication BASIC header.

**Comment by anamnavi** [comment]
@Skoucail thanks for reaching out, could you share the URL of the repository (you can remove the server name). This will allow us to check for this scenario more thoroughly. If you don't feel comfortable sharing it publicly you can email us at psgallery@microsoft.com

**Comment by Skoucail** [comment]
@anamnavi 
The url:
`https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json`
I just replaced the real domain name for 'domainname'.
If you need any more info i'm happy to help, spend almost the whole weekend trying to get this to work.

**Comment by anamnavi** [comment]
@Skoucail thanks for sending this! Could you also send the specific command you are using for the install? Including the package name you are typing in for this command and whether it includes a wildcard.

Could you also try using `Find-PSResource <pkgName> -Repository <repoWithThisUrl>` and let us know if that works.

**Comment by Skoucail** [comment]
@anamnavi 

I first register the URL with
`Register-PSResourceRepository -Name "TestRepo" -Uri "https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json" -Trusted;`

Then i try the install with (No wildcards)
`Install-PSResource -Name 'TestPackage' -Repository "TestRepo" -Verbose;`
Getting the error
`VERBOSE: Error retrieving resource from repository: Unable to load the service index for source https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json.`

When trying
`Find-PSResource "TestPackage" -Repository "TestRepo"`
I get the same error
`VERBOSE: Error retrieving resource from repository: Unable to load the service index for source https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json.`

And in the network monitoring i see 18 calls to the URL.
Basicly looks like a loop of 6 times doing the same calls.
All returning 401
Call 1:
```
GET https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json HTTP/1.1
Host: devops.domainname.be
X-NuGet-Session-Id: 1ae9e33c-b4cc-4bbc-accf-96c7fd450cc8
User-Agent: NuGet Client V3/6.2.2
X-NuGet-Client-Version: 6.2.2
Accept-Language: en-US
Accept-Encoding: gzip, deflate
```
Call 2:
```
GET https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json HTTP/1.1
Host: devops.domainname.be
X-NuGet-Session-Id: 1ae9e33c-b4cc-4bbc-accf-96c7fd450cc8
User-Agent: NuGet Client V3/6.2.2
X-NuGet-Client-Version: 6.2.2
Accept-Language: en-US
Accept-Encoding: gzip, deflate
Authorization: Negotiate TlRMTVNTUAABAAAAl4II4gAAAAAAAAAAAAAAAAAAAAAKAGFKAAAADw==
```
Call 3:
```
GET https://devops.domainname.be/DeployCollection/08c8b001-ebd5-4661-9113-40714a5a64ef/_packaging/PowerShellGallery/nuget/v3/index.json HTTP/1.1
Host: devops.domainname.be
X-NuGet-Session-Id: 1ae9e33c-b4cc-4bbc-accf-96c7fd450cc8
User-Agent: NuGet Client V3/6.2.2
X-NuGet-Client-Version: 6.2.2
Accept-Language: en-US
Accept-Encoding: gzip, deflate
Authorization: Negotiate TlRMTVNTUAADAAAAGAAYAH4AAAB2AXYBlgAAAAYABgBYAAAABgAGAF4AAAAaABoAZAAAABAAEAAMAgAAFYKI4goAYUoAAAAP2Hhbcb50DbFJy51UbdXwH0gATQBTAGoAZABjAEMASQBWAC0ARABFAFMASwBUAE8AUAAtADkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAzPLh+E9wE4ziGi2INec+NgEBAAAAAAAAqrXVTWk62QGDLjqYsXKLJAAAAAACAA4AQQBMAFMASQBDAEEARAABABYAQQBMAFMARABFAFAATABPAFkAMAAxAAQAIABhAGwAcwBpAGMAYQBkAC4AYQBsAHMAaQBjAC4AYgBlAAMAOABBAEwAUwBEAEUAUABMAE8AWQAwADEALgBhAGwAcwBpAGMAYQBkAC4AYQBsAHMAaQBjAC4AYgBlAAUAIABhAGwAcwBpAGMAYQBkAC4AYQBsAHMAaQBjAC4AYgBlAAcACACqtdVNaTrZAQYABAACAAAACAAwADAAAAAAAAAAAQAAAAAgAADA/2Ioh8KdaSRUwzt/AQdxWuUpMRF+V98xV8Emo86D3goAEAD0WmOBboOLxkU93/Lq0XrLCQAyAEgAVABUAFAALwBhAGwAcwBkAGUAcABsAG8AeQAwADEALgBhAGwAcwBpAGMALgBiAGUAAAAAAAAAAACqNUbL4B5twaCIw8fzJfk7
```

**Comment by Skoucail** [comment]
@anamnavi
Any updates/insights for this issue?

**Comment by alerickson** [comment]
This should be resolved with PR #1167, if not feel free to comment here and we can reopen this issue.
